common/src/main/java/org/apache/falcon/entity/parser/ClusterEntityParser.java

```diff
@@ -243,37 +244,72 @@ private void validateLocations(Cluster cluster) throws ValidationException {
-                        "staging".equals(locationName)
-                                ? HadoopClientFactory.ALL_PERMISSION
-                                : HadoopClientFactory.READ_EXECUTE_PERMISSION);
+                checkPathOwnerAndPermission(cluster.getName(), stagingLocation.getPath(), fs,
+                        HadoopClientFactory.READ_EXECUTE_PERMISSION, false);
             } catch (IOException e) {
-                throw new ValidationException("Unable to validate the location of name: "
-                        + location.getName() + " with path:" + location.getPath()
-                        + " for cluster " + cluster.getName(), e);
+                throw new ValidationException(
+                        "Unable to validate the location of name: " + stagingLocation.getName().value() + " with path: "
+                                + stagingLocation.getPath() + " for cluster " + cluster.getName(), e);
+            }
+            if (ClusterHelper.getLocation(cluster, ClusterLocationType.WORKING) == null) {
+                //Creating location type of working in the sub dir of staging dir with perms 755. FALCON-910
+                Location workingLocation = new Location();
+                workingLocation.setName(ClusterLocationType.WORKING);
+                String workingDirPath = stagingLocation.getPath();
+                workingDirPath = (workingDirPath.endsWith("/")
+                        ? workingDirPath.concat("working")
+                        : workingDirPath.concat("/working"));
+                workingLocation.setPath(workingDirPath);
+                try {
+                    HadoopClientFactory
+                            .mkdirs(fs, new Path(workingDirPath), HadoopClientFactory.READ_EXECUTE_PERMISSION);
+                    cluster.getLocations().getLocations().add(workingLocation);
+                } catch (IOException e) {
+                    throw new ValidationException(
+                            "Unable to create path for " + workingLocation.getName().value() + " with path: "
+                                    + workingLocation.getPath() + " for cluster " + cluster.getName(), e);
+                }
+            } else {
+                Location workingLocation = ClusterHelper.getLocation(cluster, ClusterLocationType.WORKING);
+                if (stagingLocation.getPath().equals(workingLocation.getPath())) {
+                    throw new ValidationException(
+                            "Location with name: " + stagingLocation.getName().value() + " and " + workingLocation
+                                    .getName().value() + " cannot of same path: " + stagingLocation.getPath()
+                                    + " for cluster :" + cluster.getName());
+                } else {
+                    try {
+                        checkPathOwnerAndPermission(cluster.getName(), workingLocation.getPath(), fs,
```

Comment:
checkPathOwnerAndPermission() should catch IOException and throw FalconException. IOException is a transient failure and can't be ValidationException